python/tvm/relay/ir_pass.py

```diff
@@ -516,9 +516,11 @@ def to_anf(expr, mod=None):
     return _ir_pass.to_anf(expr, mod)
 
 
-def gradient(expr, mod=None):
+def gradient(expr, mod=None, mode='higher_order'):
```

Comment:
change `mode='higher_order'` to `first_order=False`, boolean argument is easier to use than string

Reply:
there might be more moad. for example i have been thinking of adding checkpointing to relay.

Reply:
then maybe we should use enum/integer

Reply:
nvm, let's use string